src/core/Akka.Remote/Transport/Helios/HeliosTcpTransport.cs

```diff
@@ -110,7 +110,9 @@ protected override void OnMessage(NetworkData data, IConnection responseChannel)
         /// <param name="erroredChannel">The handle to the Helios channel that errored.</param>
         protected override void OnException(Exception ex, IConnection erroredChannel)
         {
-            ChannelLocalActor.Notify(erroredChannel, new UnderlyingTransportError(ex, "Non-fatal network error occurred inside underlying transport"));
+            // Want to notify only for encoding exceptions
+            if(!(ex is HeliosConnectionException))
+                ChannelLocalActor.Notify(erroredChannel, new UnderlyingTransportError(ex, "Non-fatal network error occurred inside underlying transport"));
```

Comment:
Those exceptions get surfaced as disassociations, and the only other possible type of exception coming from Helios are the recoverable non-fatal ones, such as `CorruptedFrameException`.